packages/medusa/src/services/product-variant-inventory.ts

```diff
@@ -0,0 +1,376 @@
+import { MedusaError } from "medusa-core-utils"
+import { EntityManager, In } from "typeorm"
+import {
+  IStockLocationService,
+  IInventoryService,
+  TransactionBaseService,
+} from "../interfaces"
+import { ProductVariantInventoryItem } from "../models/product-variant-inventory-item"
+import { ProductVariantService, SalesChannelLocationService } from "./"
+import { InventoryItemDTO } from "../types/inventory"
+import { ProductVariant } from "../models"
+
+type InjectedDependencies = {
+  manager: EntityManager
+  salesChannelLocationService: SalesChannelLocationService
+  productVariantService: ProductVariantService
+  stockLocationService: IStockLocationService
+  inventoryService: IInventoryService
+}
+
+class ProductVariantInventoryService extends TransactionBaseService {
+  protected manager_: EntityManager
+  protected transactionManager_: EntityManager | undefined
+
+  protected readonly salesChannelLocationService: SalesChannelLocationService
+  protected readonly productVariantService: ProductVariantService
+  protected readonly stockLocationService: IStockLocationService
+  protected readonly inventoryService: IInventoryService
+
+  constructor({
+    manager,
+    stockLocationService,
+    salesChannelLocationService,
+    productVariantService,
+    inventoryService,
+  }: InjectedDependencies) {
+    // eslint-disable-next-line prefer-rest-params
+    super(arguments[0])
+
+    this.manager_ = manager
+    this.salesChannelLocationService = salesChannelLocationService
+    this.stockLocationService = stockLocationService
+    this.productVariantService = productVariantService
+    this.inventoryService = inventoryService
+  }
+
+  async confirmInventory(
+    variantId: string,
+    quantity: number,
+    options: { sales_channel_id?: string | null } = {}
```

Comment:
suggestion: To be consistent, we could rename this to `context`. Also, should be camelCased.